### PR TITLE
argsort.

### PR DESCRIPTION
**Pull request by ueshin**

### CODE REVIEW

**Comment by itholic** [comment]
Looks way more clearer than before.
Thanks, @ueshin !! :D

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -5231,80 +5231,37 @@ def argsort(self) -> "Series":
...
-        # |                              0|                0|  3|
-        # |                              1|                1|  3|
-        # |                              2|                2|  4|
-        # |                              3|                3|  1|
-        # |                              4|                4|  6|
-        # +-------------------------------+-----------------+---+
-
-        sdf = sdf.sort(column_name).select(
-            scol_for(sdf, seq_col_name).alias(column_name)  # type: ignore
+        notnull = self.loc[self.notnull()]
```
Comment (itholic) [inline databricks/koalas/series.py:5234]: This line is that one I never thought about. Great !

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -5231,80 +5231,37 @@ def argsort(self) -> "Series":
...
+
+        sdf = sdf_for_index.join(sdf_for_data, on=tmp_join_key).drop(tmp_join_key)
+
+        internal = self._internal.with_new_sdf(
+            spark_frame=sdf, data_columns=[SPARK_DEFAULT_SERIES_NAME]
         )
+        kser = first_series(DataFrame(internal))
 
-        return first_series(DataFrame(internal))
+        return ks.concat([kser, self.loc[self.isnull()].spark.transform(lambda _: F.lit(-1))])
```
Comment (itholic) [inline databricks/koalas/series.py:5264]: And this, too. 😺

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -5231,80 +5231,37 @@ def argsort(self) -> "Series":
...
-            column_labels=[self._column_label],
-            data_spark_columns=[scol_for(sdf, column_name)],
+        tmp_join_key = verify_temp_column_name(sdf_for_data, "__tmp_join_key__")
+        sdf_for_data = InternalFrame.attach_distributed_sequence_column(sdf_for_data, tmp_join_key)
+
+        sdf = sdf_for_index.join(sdf_for_data, on=tmp_join_key).drop(tmp_join_key)
+
+        internal = self._internal.with_new_sdf(
+            spark_frame=sdf, data_columns=[SPARK_DEFAULT_SERIES_NAME]
         )
```
Comment (itholic) [inline databricks/koalas/series.py:5261]: +10

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -5231,80 +5231,37 @@ def argsort(self) -> "Series":
...
-        # sdf:
-        # +-------------------------------+-----------------+---+
-        # |__distributed_sequence_column__|__index_level_0__|  0|
-        # +-------------------------------+-----------------+---+
-        # |                              0|                0|  3|
-        # |                              1|                1|  3|
-        # |                              2|                2|  4|
-        # |                              3|                3|  1|
-        # |                              4|                4|  6|
-        # +-------------------------------+-----------------+---+
```
Comment (ueshin) [inline databricks/koalas/series.py:5260]: Although I removed these comments while I was modifying, actually I like these. Could you add the same things with a new code in the original PR?

Comment (itholic) [reply databricks/koalas/series.py:5260]: Oh, yeah. I'll add it

**Pull request merged**